module/documents/chat-message.mjs

```diff
@@ -55,6 +55,20 @@ export default class CrucibleChatMessage extends ChatMessage {
     await CrucibleAction.confirmMessage(this, {action});
   }
 
+  /* -------------------------------------------- */
+
+  /**
+   * Returns the most recent action (confirmed or otherwise) in the chat log
+   * @returns {CrucibleAction|null}
+   */
+  static getLastAction() {
+    for ( const message of game.messages.contents.toReversed() ) {
```

Comment:
`game.messages.contents.toReversed()` creates the array twice which is a bit wasteful. Better to create it once and then iterate backwards:

```suggestion
    const messages = game.messages.contents;
    for ( let i=messages.length-1; i>=0; i-- ) {
      const message = messages[i];
```